pkg/controller/proxycontroller.go

```diff
@@ -0,0 +1,74 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/cluster-olm-operator/pkg/clients"
+)
+
+func NewProxyController(name string, proxyClient *clients.ProxyClient, operatorClient *clients.OperatorClient, eventRecorder events.Recorder) factory.Controller {
```

Comment:
is it worth building out a unit test for this fella?

Reply:
There are admittedly few tests for the controllers, proper.